### ISSUE TITLE
Allowing RN 0.74 as a peer dependency

### ISSUE THREAD

**Issue by naftalibeder** [react-native-clipboard/clipboard#239]
Would you accept a PR updating the peer dependencies to include `"react-native": "...  || ^0.74.0"`? The library works fine on 0.74, but throws an error on dependency installation at the moment.

**Comment by sargsyanapach94** [comment]
I have the same issue.

**Comment by davidgvf** [comment]
Same problem

**Comment by cixio** [comment]
```
npm update

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: App@1.0.1
npm ERR! Found: react-native@0.74.0
npm ERR! node_modules/react-native
npm ERR!   react-native@"0.74.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react-native@"^0.61.5 || ^0.62.3 || ^0.63.2 || ^0.64.2 || ^0.65.0 || ^0.66.0 || ^0.67.0 || ^0.68.0 || ^0.69.0 || ^0.70.0 || ^0.71.0 || ^0.72.0 || ^0.73.0" from @react-native-clipboard/clipboard@1.14.1
npm ERR! node_modules/@react-native-clipboard/clipboard
npm ERR!   @react-native-clipboard/clipboard@"^1.14.1" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by codetobuild** [comment]
> How I fixed in the interim for my app... add dependency version `overrides` in my app's `package.json`:
> 
> ```
>   "overrides": {
>     "@react-native-clipboard/clipboard": {
>       "react-native": "~0.74.0",
>       "react-native-windows": "~0.74.0"
>     }
>   },
> ```
This code made my day :)

**Comment by MuhammadRafeh** [comment]
Any fix?

**Comment by ToniNikolaev23** [comment]
I have this issue also with react-native: 0.71.2.

**Comment by OmarThinks** [comment]
I have the same issue on React Native 0.72.14

```bash
$ npm install --save @react-native-clipboard/clipboard
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: Awesome@0.0.1
npm error Found: react-native@0.72.14
npm error node_modules/react-native
npm error   react-native@"0.72.14" from the root project
npm error   peer react-native@"^0.61.5 || ^0.62.3 || ^0.63.2 || ^0.64.2 || ^0.65.0 || ^0.66.0 || ^0.67.0 || ^0.68.0 || ^0.69.0 || ^0.70.0 || ^0.71.0 || ^0.72.0 || ^0.73.0" from @react-native-clipboard/clipboard@1.14.1
npm error   node_modules/@react-native-clipboard/clipboard
npm error     @react-native-clipboard/clipboard@"*" from the root project
npm error
npm error Could not resolve dependency:
npm error peer react-native@"^0.73.0" from react-native-windows@0.73.13
npm error node_modules/react-native-windows
npm error   peer react-native-windows@"^0.61.0 || ^0.62.0 || ^0.63.0 || ^0.64.0 || ^0.65.0 || ^0.66.0 || ^0.67.0 || ^0.68.0 || ^0.69.0 || ^0.70.0 || ^0.71.0 || ^0.72.0 || ^0.73.0" from @react-native-clipboard/clipboard@1.14.1
npm error   node_modules/@react-native-clipboard/clipboard
npm error     @react-native-clipboard/clipboard@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

I guess it's on all React Native versions

**Comment by EducoreAPK** [comment]
Same issue !!!

**Comment by LPitonakova** [comment]
After the change, the npm package is installed ok but when running `pod install --repo-update`, I get

```
 Invalid `RNCClipboard.podspec` file: undefined method `visionos' for #<Pod::Specification name="RNCClipboard">
```

**Comment by lchenfox** [comment]
```
npm notice
npm notice New minor version of npm available! 10.7.0 -> 10.8.1
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.8.1
npm notice To update run: npm install -g npm@10.8.1
npm notice
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: jye@0.0.1
npm error Found: react-native@0.72.0
npm error node_modules/react-native
npm error   react-native@"0.72.0" from the root project
npm error   peer react-native@"^0.61.5 || ^0.62.3 || ^0.63.2 || ^0.64.2 || ^0.65.0 || ^0.66.0 || ^0.67.0 || ^0.68.0 || ^0.69.0 || ^0.70.0 || ^0.71.0 || ^0.72.0 || ^0.73.0" from @react-native-clipboard/clipboard@1.14.1
npm error   node_modules/@react-native-clipboard/clipboard
npm error     @react-native-clipboard/clipboard@"^1.11.1" from the root project
npm error
npm error Could not resolve dependency:
npm error peer react-native@"^0.73.0" from react-native-windows@0.73.15
npm error node_modules/react-native-windows
npm error   peer react-native-windows@"^0.61.0 || ^0.62.0 || ^0.63.0 || ^0.64.0 || ^0.65.0 || ^0.66.0 || ^0.67.0 || ^0.68.0 || ^0.69.0 || ^0.70.0 || ^0.71.0 || ^0.72.0 || ^0.73.0" from @react-native-clipboard/clipboard@1.14.1
npm error   node_modules/@react-native-clipboard/clipboard
npm error     @react-native-clipboard/clipboard@"^1.11.1" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /Users/chenlong/.npm/_logs/2024-06-04T02_39_56_054Z-eresolve-report.txt

npm error A complete log of this run can be found in: /Users/chenlong/.npm/_logs/2024-06-04T02_39_56_054Z-debug-0.log
```

Also face the same issue.

**Comment by GoldenBilly** [comment]
okay, ill just use --force argument then

**Comment by GoldenBilly** [comment]
i know it may be not that good idea, but i need this lib

**Comment by GoldenBilly** [comment]
helped for me to remove ERESOLVE error when i use npm i in cloned repository

> How I fixed in the interim for my app... add dependency version `overrides` in my app's `package.json`:
> 
> ```
>   "overrides": {
>     "@react-native-clipboard/clipboard": {
>       "react-native": "~0.74.0",
>       "react-native-windows": "~0.74.0"
>     }
>   },
> ```

**Comment by ezescigo** [comment]
Same issue here for RN 0.71.14

**Comment by Catnais88** [comment]
Same for me

**Comment by sudhanshu-busy** [comment]
Any solutions?

**Comment by ezescigo** [comment]
> Any solutions?
> 

Until they fix it, temp fix is the override posted above

**Comment by sudhanshu-busy** [comment]
> > Any solutions?
> 
> Until they fix it, temp fix is the override posted above

how to do it pls write steps my react native version is 0.70.3

**Comment by GoldenBilly** [comment]
> > > Any solutions?
> > 
> > 
> > Until they fix it, temp fix is the override posted above
> 
> how to do it pls write steps my react native version is 0.70.3

open your project, go to your package json, add that:
```
  "overrides": {
    "@react-native-clipboard/clipboard": {
      "react-native": "~0.73.0",
      "react-native-windows": "~0.73.0"
    }
  },
```
then use `npm i @react-native-clipboard/clipboard`
(i didnt tested it, but it should help)

**Comment by sudhanshu-busy** [comment]
> > > > Any solutions?
> > > 
> > > 
> > > Until they fix it, temp fix is the override posted above
> > 
> > 
> > how to do it pls write steps my react native version is 0.70.3
> 
> open your project, go to your package json, add that:
> 
> ```
>   "overrides": {
>     "@react-native-clipboard/clipboard": {
>       "react-native": "~0.70.3",
>       "react-native-windows": "~0.70.3"
>     }
>   },
> ```
> 
> then use `npm i @react-native-clipboard/clipboard` (i didnt tested it, but it should help)

It didn't work I had tried earlier what I did is used some old version of this lib that is used in example of this project.
It installed successfully, is it fine?

**Comment by GoldenBilly** [comment]
> > > > > Any solutions?
> > > > 
> > > > 
> > > > Until they fix it, temp fix is the override posted above
> > > 
> > > 
> > > how to do it pls write steps my react native version is 0.70.3
> > 
> > 
> > open your project, go to your package json, add that:
> > ```
> >   "overrides": {
> >     "@react-native-clipboard/clipboard": {
> >       "react-native": "~0.70.3",
> >       "react-native-windows": "~0.70.3"
> >     }
> >   },
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > then use `npm i @react-native-clipboard/clipboard` (i didnt tested it, but it should help)
> 
> It didn't work I had tried earlier what I did is used some old version of this lib that is used in example of this project. It installed successfully, is it fine?

i cant say it, you should test it in your project

**Comment by GoldenBilly** [comment]
its same if we use just `npm install @react-native-clipboard/clipboard --force`

**Comment by ezescigo** [comment]
> > > > > Any solutions?
> > > > 
> > > > 
> > > > Until they fix it, temp fix is the override posted above
> > > 
> > > 
> > > how to do it pls write steps my react native version is 0.70.3
> > 
> > open your project, go to your package json, add that:
> > 
> > ```
> >   "overrides": {
> >     "@react-native-clipboard/clipboard": {
> >       "react-native": "~0.70.3",
> >       "react-native-windows": "~0.70.3"
> >     }
> >   },
> > ```
> > 
> > then use `npm i @react-native-clipboard/clipboard` (i didnt tested it, but it should help)
> 
> It didn't work I had tried earlier what I did is used some old version of this lib that is used in example of this project.
> It installed successfully, is it fine?
>  

other option is just using npm i --legacy-peer-deps

**Comment by joshticks002** [comment]
This worked for me going through the error logs on react native version 0.74.3

root/node_modules/@react-native-clipboard/clipboard/android/src/main/java/com/reactnativecommunity/clipboard/ClipboardModule.java

Edit line 241:
-  @Override
    public void removeListeners(int count) {

    }
+ @Override
    public void removeListeners(double count) {

    }

**Comment by AlexLFundbox** [comment]
Same issue

**Comment by tadokorokantoku** [comment]
Same issue

**Comment by IndraMyn** [comment]
i just run

`npm install --save @react-native-clipboard/clipboard --force`

and that's fix my problem, but i think is not best practice.
maybe we just need to wait for latest version that supported in react native 0.74^

**Comment by YashM20** [comment]
> This worked for me going through the error logs on react native version 0.74.3
> 
> newArchEnabled=true hermesEnabled=true
> 
> root/node_modules/@react-native-clipboard/clipboard/android/src/main/java/com/reactnativecommunity/clipboard/ClipboardModule.java
> 
> Edit line 241: remove @OverRide public void removeListeners(int count) {
> 
> ```
> }
> ```
> 
> add @OverRide public void removeListeners(double count) {
> 
> ```
> }
> ```

This seems to be proper solution.

Its working for me👍.
my config,
newArchEnabled=true 
hermesEnabled=true
react-native@0.74.5

**Comment by jackfiallos** [comment]
> After the change, the npm package is installed ok but when running `pod install --repo-update`, I get
> 
> ```
>  Invalid `RNCClipboard.podspec` file: undefined method `visionos' for #<Pod::Specification name="RNCClipboard">
> ```

if somebody is still experiencing this issue with podspec, just remove or comment the visionos lines (temporary solution)

```ruby
# s.visionos.source_files = "ios/**/*.{h,m,mm}"
```

**Comment by Naturalclar** [comment]
made fix in 1.14.3 :+1:
